src/tools/lib/python/energy_histogram.py

```diff
@@ -0,0 +1,197 @@
+"""
+This file is part of PIConGPU.
+
+PIConGPU is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+PIConGPU is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with PIConGPU.
+If not, see <http://www.gnu.org/licenses/>.
+
+Copyright 2017 Marco Garten, Axel Huebl
+Authors: Axel Huebl
+License: GPLv3+
+"""
+import numpy as np
+import matplotlib as mpl
+from matplotlib import pyplot as plt
+import h5py
+
+
+class EnergyHistogram:
+    """
+    Class to help plotting data from the energyHistogram plugin.
+    """
+    def __init__(
+            self,
+            species_name="e",
+            species_filter="",
+            sim="./",
+            period=1000,
+            bin_count=1024,
+            min_energy=10,
+            max_energy=10000
+        ):
+        """
+        Parameters
+        ----------
+        species_name : string
+            name of the species for which the energy histogram should be
+            plotted (default: "e")
+        species_filter : string
+            file suffix for plugin output created by a run over a subselection
+            of the species according to an optional filter argument
+            (default: "")
+        sim : string
+            path to simOutput directory (default: "./")
+        period : unsigned integer
```

Comment:
the defaults of those four variables should be `=None` and auto-detected.

Only if the user chooses to change the limits it should be required (the user can't change the resolution or periodicity anyway) after the sim